airflow/plugins_manager.py

```diff
@@ -262,6 +263,8 @@ def ensure_plugins_loaded():
 
     log.debug("Loading plugins")
 
+    start = time.monotonic()
```

Comment:
Might there be a usecase for a general timing context manager here?

```python
with timer() as t:
    # ...

log.info("Frobbing the foonars took %.2f seconds", t.elapsed.total_seconds())
```

Encapsulation makes it easier to make sure everyone gets this right (using `time.monotonic()`!), and might make it easier to automatically export metrics to statd / prometheus if desired (`with timer(metric="metric_name") as t:`).

Reply:
Could be. For metrics there is already a `with Stats.timer("my_metric_name":)` though.

Reply:
#12353

Reply:
Only problem I see is that context manager introduces additional level of indentation which can require a pylint disable sometimes... Apart from that I like the idea!

Reply:
Can also do it like this:

```python
timer = Stats.timer().start()
...
timer.end()
# use timer.duration
```